content/starter-configs/tutorials/wordpress.md

```diff
@@ -48,62 +50,106 @@ define('DB_PASSWORD', 'tugboat');
   php:
-    # Use PHP 7.x with Apache to serve the WordPress site; this syntax pulls in the latest version of PHP 7
-    image: tugboatqa/php:7-apache
+    # Use PHP 8.x with Apache to serve the WordPress site; this syntax pulls in the latest version of PHP 8.1
+    image: tugboatqa/php:8.1-apache
 
     # Set this as the default service. This does a few things
     #   1. Clones the git repository into the service container
     #   2. Exposes port 80 to the Tugboat HTTP proxy
     #   3. Routes requests to the preview URL to this service
     default: true
 
-    # Wait until the mysql service is done building
+    # Wait until the mysql service is done building so we have a database.
     depends: mysql
 
-    # A set of commands to run while building this service
+    # A set of commands to run while building this service.
     commands:
-      # Commands that set up the basic preview infrastructure
+      # Phase 1 (init): Configure the server infrastructure.
       init:
-        # Install prerequisite packages
+        # Install prerequisite packages.
         - apt-get update
-        - apt-get install -y rsync libzip-dev
+        - apt-get install -y rsync
 
-        # Enable extra apache modules.
-        - a2enmod rewrite headers
+        # Turn on URL rewriting.
+        - a2enmod rewrite
 
-        # Install the PHP extensions
-        - docker-php-ext-install mysqli exif zip
-        # Install imagick
-        - apt-get install -y libmagickwand-dev
-        - pecl install imagick-beta -y
-        - docker-php-ext-enable imagick
+        # Install PHP mysqli extension.
+        - docker-php-ext-install mysqli
 
-        # Install wp-cli
+        # Install the Wordpress CLI tool.
         - curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
         - chmod +x wp-cli.phar
         - mv wp-cli.phar /usr/local/bin/wp
 
-        # Use the tugboat-specific wp-config.local.php
-        - cp "${TUGBOAT_ROOT}/.tugboat/wp-config.local.php" "${DOCROOT}/"
+        # If using composer to install Wordpress, uncomment this line.
+        # - composer install --optimize-autoloader
 
-        # Link the document root to the expected path. This example links /web
-        # to the docroot
-        - ln -snf "${TUGBOAT_ROOT}/web" "${DOCROOT}"
+        # Link Wordpress core as the docroot.
+        - ln -snf "${TUGBOAT_ROOT}/docroot" "${DOCROOT}"
 
-      # Commands that import files, databases, or other assets. When an
-      # existing preview is refreshed, the build workflow starts here,
-      # skipping the init step, because the results of that step will
-      # already be present.
+        # Copy and link the Tugboat config file into the Wordpress core directory.
+        - rm -rf ${TUGBOAT_ROOT}/wp-config.php; rm -rf ${DOCROOT}/wp-config.php; cp
+          ${TUGBOAT_ROOT}/.tugboat/wp-config.tugboat.php ${TUGBOAT_ROOT}/wp-config.php
+        - ln -snf ${TUGBOAT_ROOT}/wp-config.php ${DOCROOT}/wp-config.php
```

Comment:
Is there a reason we are placing this file both at the Tugboat root and the docroot?

Reply:
This would create problems if core is the root directory of the repo, which is another common pattern for Wordpress.

Reply:
I know I had a reason for this...something about symlinking and keeping files out of the repo.  But I suppose in this case we can simply overwrite the original `wp-config.php` completely no matter how it's setup.

The one drawback here is that you can't include any sub-config files into the main config file to keep any other secret info out of the repo.  With a little trickery, though, you could load them into the Tugboat UI as environment vars, then use the WP CLI tool in the tugboat config to inject them into your site.